pkg/controller/issueapi/handle_issue.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package issueapi
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/google/exposure-notifications-verification-server/pkg/api"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller"
+	"github.com/google/exposure-notifications-verification-server/pkg/database"
+	"github.com/google/exposure-notifications-verification-server/pkg/observability"
+
+	"go.opencensus.io/tag"
+)
+
+type issueResult struct {
+	httpCode    int
+	errorReturn *api.ErrorReturn
+	obsBlame    tag.Mutator
+	obsResult   tag.Mutator
+}
+
+func (c *Controller) HandleIssue() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if c.config.IsMaintenanceMode() {
+			c.h.RenderJSON(w, http.StatusTooManyRequests,
+				api.Errorf("server is read-only for maintenance").WithCode(api.ErrMaintenanceMode))
+			return
+		}
+
+		ctx := observability.WithBuildInfo(r.Context())
+
+		result := &issueResult{
+			httpCode:  http.StatusOK,
+			obsBlame:  observability.BlameNone,
+			obsResult: observability.ResultOK(),
+		}
+		defer recordObservability(&ctx, result)
+
+		var request api.IssueCodeRequest
+		if err := controller.BindJSON(w, r, &request); err != nil {
+			result.obsBlame = observability.BlameClient
+			result.obsResult = observability.ResultError("FAILED_TO_PARSE_JSON_REQUEST")
+			c.h.RenderJSON(w, http.StatusBadRequest, api.Error(err))
+			return
+		}
+
+		authApp, user, err := c.getAuthorizationFromContext(r)
+		if err != nil {
+			result.obsBlame = observability.BlameClient
+			result.obsResult = observability.ResultError("MISSING_AUTHORIZED_APP")
+			c.h.RenderJSON(w, http.StatusUnauthorized, api.Error(err))
+			return
+		}
+
+		var realm *database.Realm
```

Comment:
I see this pattern in a few places and it confuses me. The [middleware sets the realm from the authApp](https://github.com/google/exposure-notifications-verification-server/blob/eff1dd4c5e9c4cd639cca0fc8bebb0de36d4274f/pkg/controller/middleware/apikey.go#L111) already, and it's not possible to arrive at this page without that middleware running.

Reply:
I agree. This is just modelled after what's there, but some of the controllers seem to double-check (with what is unreachable code). Maybe its defensive or was written before middleware? Either way let's simplify